### DISCUSSION TITLE
Vercel should expose an environment variable for the Git branch specific URL

### DISCUSSION THREAD

**Discussion by ruohola** [vercel/community#217] [Idea]
Currently the only URL environment variable that Vercel exposes for running apps is `VERCEL_URL` (in the format `<project-name>-<unique-hash>-<scope-slug>.vercel.app`) which links to the unique id of that one deployment (this is the so called [_automatic deployment URL_](https://vercel.com/docs/concepts/deployments/automatic-urls#)).

But the URL that the Vercel bot links in GitHub pull request's comments for preview enviroments is not this same deployment URL, but instead the branch specific URL (this is the so called [_automatic branch URL_](https://vercel.com/docs/concepts/deployments/automatic-urls#automatic-branch-urls)).

This is problematic, since now in one's code there is no way to programmatically access this branch specific URL which is the actual URL that the preview application is actually most likely accessed from.

Suggestion:

Expose a `VERCEL_BRANCH_URL` environment variable during runtime (and `NEXT_PUBLIC_VERCEL_BRANCH_URL` etc. during build time) which contain the URL for the branch, in the format: `<project-name>-git-<branch-name>-<scope-slug>.vercel.app`.

**Comment by fvpDev** [comment]
Or something close to `VERCEL_GIT_REPO_SLUG`...maybe `VERCEL_GIT_REPO_BRANCH`. It would be particularly useful for having multiple versions of a site based on the git branch, exposing certain components based on what version of the site is being viewed.

**Reply by fvpDev** [reply]
Ok, so after some console logging I found the variable actually. It's called `VERCEL_GIT_COMMIT_REF`. I dare say, it's not an intuitive variable name for the value that it actually exposes. I think it should be changed to `VERCEL_GIT_REPO_BRANCH`.
